cmd/virt-prerunner/main.go

```diff
@@ -45,9 +46,17 @@ func main() {
 	if vmConfig.Cmdline != nil {
 		cloudHypervisorCmd = append(cloudHypervisorCmd, "--cmdline", fmt.Sprintf("'%s'", vmConfig.Cmdline.Args))
 	}
-	cloudHypervisorCmd = append(cloudHypervisorCmd, "--cpus", fmt.Sprintf("boot=%d,topology=%d:%d:%d:%d",
+	vcpuToPCPU := []string{}
+	for _, affinity := range vmConfig.Cpus.Affinity {
+		vcpuToPCPU = append(vcpuToPCPU, fmt.Sprintf("%d@[%d]", affinity.Vcpu, affinity.HostCpus[0]))
+	}
+	cpuAffinity := ""
+	if len(vcpuToPCPU) > 0 {
```

Comment:
if `len(vcpuToPCPU)` is 0, `cpuAffinity` could be empty, is it OK for the `affinity=%s` part?

Reply:
It's ok to give an empty value to `affinity`.